packages/assertion-monitor/index.ts

```diff
@@ -0,0 +1,193 @@
+    },
+    {
+      type: 'tuple',
+      name: 'assertion',
+      components: [
+        {
+          type: 'tuple',
+          name: 'beforeState',
+          components: [
+            {
+              type: 'tuple',
+              name: 'globalState',
+              components: [
+                {
+                  type: 'bytes32[2]',
+                  name: 'bytes32Vals',
+                },
+                {
+                  type: 'uint64[2]',
+                  name: 'u64Vals',
+                },
+              ],
+            },
+            {
+              type: 'uint8',
+              name: 'machineStatus',
+            },
+          ],
+        },
+        {
+          type: 'tuple',
+          name: 'afterState',
+          components: [
+            {
+              type: 'tuple',
+              name: 'globalState',
+              components: [
+                {
+                  type: 'bytes32[2]',
+                  name: 'bytes32Vals',
+                },
+                {
+                  type: 'uint64[2]',
+                  name: 'u64Vals',
+                },
+              ],
+            },
+            {
+              type: 'uint8',
+              name: 'machineStatus',
+            },
+          ],
+        },
+        {
+          type: 'uint64',
+          name: 'numBlocks',
+        },
+      ],
+      indexed: false,
+    },
+    {
+      type: 'bytes32',
+      name: 'afterInboxBatchAcc',
+      indexed: false,
+    },
+    {
+      type: 'bytes32',
+      name: 'wasmModuleRoot',
+      indexed: false,
+    },
+    {
+      type: 'uint256',
+      name: 'inboxMaxCount',
+      indexed: false,
+    },
+  ],
+} as const
+
+const getLogsForLast7Days = async (
+  client: PublicClient,
+  address: `0x${string}` | `0x${string}`[],
+  abi: AbiEvent
+) => {
+  const latestBlockNumber = await client.getBlockNumber()
+  const oneWeekInSeconds = 7 * 24 * 60 * 60
+  const earliestBlock = await client.getBlock({
+    blockNumber: latestBlockNumber - BigInt(oneWeekInSeconds / 12),
+  }) // assuming average block time of 12 seconds
+
+  return await client.getLogs({
+    address,
+    event: nodeCreatedEventAbi,
+    fromBlock: earliestBlock.number,
+    toBlock: latestBlockNumber,
+  })
+}
+
+const monitorNodeCreatedEvents = async (childChainInformation: ChainInfo) => {
+  const parentChain = getChainFromId(childChainInformation.parentChainId)
+  const client = createPublicClient({
+    chain: parentChain,
+    transport: http(childChainInformation.parentRpcUrl),
+  }) as any
+
+  const logs = await getLogsForLast7Days(
+    client,
+    childChainInformation.ethBridge.rollup as `0x${string}`,
+    nodeCreatedEventAbi
+  )
+
+  if (!logs || logs.length === 0) {
+    const alertMessage = `No assertion events found on ${childChainInformation.name} in the last 7 days.`
+    console.error(alertMessage)
+    if (options.enableAlerting) {
+      await reportAssertionMonitorErrorToSlack({ message: alertMessage })
```

Comment:
This will ping us on slack N times for N chains that raise the alert, leading to a spam-like experience. A better approach here would be to store all the alerts in a temporary array, and later, if length > 0, we log 1 final summary that contains all the alerts in 1 message. (This has already been achieved in batch-poster-monitor)